### ISSUE TITLE
Root and ultimate ancestor oddities

### ISSUE THREAD

**Issue by hyanwong** [tskit-dev/tsinfer#275]
Investigating the tree sequences produced by the latest tsinfer version reveals some oddities in the two oldest ancestors. Now that times are set as *frequencies* by default, it seems rather odd that all; the ancestors are at times 0...1 except the 2 oldest, which are at times ~2.0 and ~3.0, because we insert a standard root at max_time+1 and an "ultimate ancestor" at time max_time+2. Since it's only the relative time that is important, this shouldn't be a problem, but it might be helpful (e.g. for plotting reasons) if the max_ancestor_time is < 1, to set the root_time to 1.

Additionally, in https://github.com/tskit-dev/tsinfer/blob/master/tsinfer/inference.py#L587 the comments say

"Add the ultimate ancestor. This is an awkward hack really; we don't ever insert this ancestor."

I suggest that if this ancestor is never expected to be inserted, we should set the time to `+inf`, so that if it *is* inserted we get an error. I did this, and it turn out it *is* inserted and trust up in the final tree sequence, for instance:

```
ts = msprime.simulate(10, recombination_rate=1, mutation_rate=2, random_seed=123)
sd = tsinfer.SampleData.from_tree_sequence(ts, use_times=False)
anc = tsinfer.generate_ancestors(sd)
print("Ultimate time (should not be used):", anc.ancestors_time[0])
print("Universal root (should be oldest):", anc.ancestors_time[1])

a_ts =  tsinfer.match_ancestors(sd, anc)
infer_ts =  tsinfer.match_samples(sd, a_ts)
print("Oldest time in ts is:", infer_ts.max_root_time)
```

giving

```
Ultimate time (should not be used): 2.8
Universal root (should be oldest): 1.8
Oldest time in ts is: 2.8
```

**Comment by jeromekelleher** [comment]
Yes, the oldest ancestor stuff is hacky. We should fix this for the next release.

**Comment by jeromekelleher** [comment]
Pushing this back to the 0.3 series, the ultimate ancestor assumption is buried deep in the guts of the LS algorithm and it's not worth messing with it now for 0.2.

**Comment by hyanwong** [comment]
Just to note - I suspect that the oldest root is present in the final inferred ts because when we simplify we now `keep_unary=True`. Previously it was simplified out.

**Comment by jeromekelleher** [comment]
That's a good point.

**Comment by hyanwong** [comment]
I'm unclear if the ultimate ancestor is to "make sure that the ancestor IDs we have in the ancestor file are the same as in the ancestor tree sequence" (as in the comment) or "buried deep in the guts of the LS algorithm"? Perhaps both? Since we are keeping track of the ancestors data ID in the metadata, I wonder if it would be easy to remove or not? If used in the LS algorithm, then presumably not, as the metadata is injected at the end.

We could, perhaps, simply remove the ultimate ancestor from the final tree sequence (e.g. by removing any node where `metadata["ancestor_data_id"] == 0`?

**Comment by jeromekelleher** [comment]
There's two ultimate ancestors here - one we *definitely* need, representing the haplotype of all zeros everywhere. LS won't work without this.

The second one is this ancestors parent, which I think we put in just as a simple way to make IDs match. So, I think we could remove the call where we add this second ancestor and just fix the test suite breakage.

**Comment by hyanwong** [comment]
Thanks for the clarification. I completely get the first one (I think we call this the "root with zeros at every position"). I agree that it sounds like we can just remove the second one.
